### PR TITLE
Add test/ to pylint and flake

### PR DESCRIPTION
**Pull request by dc-dc-dc**
Closes: #581

### CODE REVIEW

File: test/external_test_llvm.py [hunk 1]
```diff
@@ -49,12 +49,12 @@ def test_add_sum_add(self):
 if __name__ == "__main__":
   unittest.main()
 
-  """
```
Comment (geohot) [inline test/external_test_llvm.py:52]: Why is this bad, aren't quote strings fine? Tbh, these should probably be deleted.

File: test/test_ops.py [hunk 2]
```diff
@@ -236,22 +236,22 @@ def test_expand(self):
   def test_sd_big_conv(self):
     # internal shape (1, 1, 512, 62, 62, 512, 3, 3) overflows a int
     helper_test_op([(1,256,64,64), (512,256,3,3)],
-                    lambda x,w: torch.nn.functional.conv2d(x, w),
+                    torch.nn.functional.conv2,
```
Comment (geohot) [inline test/test_ops.py:239]: broken

File: test/test_ops.py [hunk 3]
```diff
@@ -403,8 +403,8 @@ def test_padded_conv2d(self):
     for p in [2, (2,1), (2,2)]:
       with self.subTest(padding := p):
         helper_test_op([(bs,cin,11,28), (4,cin,H,W)],
-          lambda x,w: torch.nn.functional.conv2d(x,w,padding=padding).relu(),
-          lambda x,w: Tensor.conv2d(x,w,padding=padding).relu(), atol=1e-4)
+          lambda x,w, padding=padding: torch.nn.functional.conv2d(x,w,padding=padding).relu(),
```
Comment (geohot) [inline test/test_ops.py:406]: Why can't it capture?

Comment (dc-dc-dc) [reply test/test_ops.py:406]: lint complains that the variable is from the loop, can override by passing and setting the default

File: test/test_ops.py (hunk 2 above)
Comment (dc-dc-dc) [reply test/test_ops.py:239]: this functions the same, and satisfies the linter check for unnecessary lambda

File: test/external_test_llvm.py (hunk 1 above)
Comment (dc-dc-dc) [reply test/external_test_llvm.py:52]: the linter considers it a string statement in this case, if it was under a function it would be fine

File: test/test_ops.py (hunk 2 above)
Comment (geohot) [reply test/test_ops.py:239]: no it doesn't, please read carefully

Comment (dc-dc-dc) [reply test/test_ops.py]: typo, fixed it and added a conditional to the the skip for ALL env so it can easily be run without having to call the function directly

File: test/test_speed_v_torch.py [hunk 4]
```diff
@@ -221,13 +221,12 @@ def test_conv2d(self):
...
           torch_dat = torch.rand(bs, in_chans, img_size, img_size).to(torch_device)
           torch_conv = torch.nn.Conv2d(in_chans, out_chans, 3, bias=None).to(torch_device)
-
           tiny_dat = Tensor(torch_dat.cpu().numpy())
           tiny_conv = Conv2d(in_chans, out_chans, 3, bias=None)
           tiny_conv.weight = Tensor(torch_conv.weight.detach().cpu().numpy())
 
-          def f1(torch_dat): return torch_conv(torch_dat)
-          def f2(tiny_dat): return tiny_conv(tiny_dat).realize()
+          def f1(td, torch_conv=torch_conv): return torch_conv.__call__(td)
```
Comment (geohot) [inline test/test_speed_v_torch.py:228]: Why is this needed here and not on line 212?

Comment (dc-dc-dc) [reply test/test_speed_v_torch.py:228]: the linter is checking for when variables in closures are defined in loops

**Comment by geohot** [comment]
I've been reluctant to merge this because linting is supposed to make code better quality, and I feel that this merge makes it worse, with a bunch of hacks to just satisfy the linter. Either that code should be rewritten in a better style, or those linting rules should be disabled for test/.

Also, add the test lint to the precommit hooks.

**Comment by dc-dc-dc** [comment]
I think the unnecessary lambdas were fine and cleaning up unnecessary imports, I agree that the var from loop hack was more for linter than better code, one of the rules to disable for test folder?

**Comment by geohot** [comment]
Want to create just a PR with the positive cleanups and no linter yet? I agree on the lambdas

**Comment by dc-dc-dc** [comment]
yeah, will close this one and open a clean pr
